contracts/CollectiveGovernance.sol

```diff
@@ -552,13 +635,68 @@ contract CollectiveGovernance is Governance, VoteStrategy, ERC165 {
         return _communityDescription;
     }
 
+    function castVoteFor(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.voteForByShare(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteCount(_proposalId, msg.sender, _tokenId, count, 0);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function castVoteAgainst(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.voteAgainstByShare(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteCount(_proposalId, msg.sender, _tokenId, 0, count);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function castAbstention(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.abstainForShare(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteCount(_proposalId, msg.sender, _tokenId, 0, 0);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function _undoVote(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.undoVoteById(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteUndo(_proposalId, msg.sender, count);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function sendGasRebate(address recipient, uint256 startGas) internal {
```

Comment:
what's the reason for this being internal and not private?

Reply:
Of course it can be private.   When I thought about it for a few seconds I considered that any future contract that extends this one might also want to send refunds. 

> what's the reason for this being internal and not private?